pkg/gatewayserver/gatewayserver.go

```diff
@@ -150,6 +151,12 @@ func New(c *component.Component, conf *Config) (gs *GatewayServer, err error) {
 	}
 
 	hooks.RegisterUnaryHook("/ttn.lorawan.v3.NsGs", rpclog.NamespaceHook, rpclog.UnaryNamespaceHook("gatewayserver"))
+	bsCtx := gs.FillContext(ctx)
```

Comment:
This doesn't look quite right. `FillContext` is intended to fill request contexts, not to extend the component context.

Reply:
Hmm ok. This is again from the original plumbing code.

If that's the case, then we should directly pass the component context here and use this function on the request context in the echo handler right?

cc: @johanstokking

Reply:
Yes

Reply:
FTR: no need for the above. `FillContext` is already called by `FillGatewayContext`.